lms/static/scripts/frontend_apps/components/InitialLoadingTracker.tsx

```diff
@@ -0,0 +1,76 @@
+import { Spinner } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useMemo, useRef, useState } from 'preact/hooks';
+
+import { InitialLoadingContext } from '../utils/initial-loading-context';
+
+export type InitialLoadingProps = {
+  children: ComponentChildren;
+  classes?: string | string[];
+};
+
+export default function InitialLoadingTracker({
```

Comment:
I'm open to suggestions on the name of this component 😅

Reply:
`HideUntilLoaded`? That would make it obvious what is going on when you look at the component tree.